plan_delegator/src/plan_delegator.cpp

```diff
@@ -547,9 +549,19 @@ namespace plan_delegator
 
             if(adjusted_start_dist < original_starting_lanelet_centerline_start_point_dt){
                 auto previous_lanelets = wm_->getMapRoutingGraph()->previous(original_starting_lanelet, false);
-
                 if(!previous_lanelets.empty()){
-                    setManeuverStartingLaneletId(maneuver, previous_lanelets[0].id());
+                    auto llt_on_route_optional = wm_->getFirstLaneletOnShortestPath(previous_lanelets);
+                    lanelet::ConstLanelet previous_lanelet_to_add;
+
+                    if (llt_on_route_optional){
+                        previous_lanelet_to_add = llt_on_route_optional.value();
+                    }
+                    else{
+                        RCLCPP_WARN_STREAM(rclcpp::get_logger("plan_delegator"), "When adjusting maneuver for maneuvers other than lane follow, no previous lanelet found on the shortest path. " <<
```

Comment:
Nit: We can reduce the length of this log statement with `When adjusting non-lane follow maneuver, no previous ...`

Reply:
Revised